packages/repository/test/unit/type-resolver.unit.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  isBuiltinType,
+  isTypeResolver,
+  resolveType,
+} from '../../src/type-resolver';
+
+class SomeModel {
+  constructor(public name: string) {}
+}
+
+const A_DATE_STRING = '2018-01-01T00:00:00.000Z';
+
+describe('isTypeResolver', () => {
+  it('returns false when the arg is a class', () => {
+    expect(isTypeResolver(SomeModel)).to.be.false();
+  });
+
+  it('returns false when the arg is not a function', () => {
+    expect(isTypeResolver(123)).to.be.false();
+  });
+
+  it('returns false when the arg is String type', () => {
+    expect(isTypeResolver(String)).to.be.false();
+  });
+
+  it('returns false when the arg is Number type', () => {
+    expect(isTypeResolver(Number)).to.be.false();
+  });
+
+  it('returns false when the arg is Boolean type', () => {
+    expect(isTypeResolver(Boolean)).to.be.false();
+  });
+
+  it('returns false when the arg is Object type', () => {
```

Comment:
Nitpick: since the logic in these tests is the same, can you create a helper function which takes in the parameter to the `isTypeResolver` function and creates a test case with the type name as well as the expected result?

Reply:
Creating parameterized test cases is tricky. If done wrong, then test failures don't point to the source code of the individual test case, only to the same helper function. In coercion tests, we found a solution where we intercept errors and append extra stack trace to them in order to preserve location of where the test case was defined.

In this particular case, I think such extra complexity outweighs benefits.